lib/base.js

```diff
@@ -1,138 +1,69 @@
+var _ = require('lodash');
 
-var BaseElement;
+var ElementRegistry = require('./registry').ElementRegistry;
 
-/*
- * A refract element implementation with an extensible element registry.
- *
- * The element registry allows you to register your own classes to be instantiated
- * when a particular refract element is encountered, and allows you to specify
- * which elements get instantiated for existing Javascript objects.
- */
-
-var ElementRegistry = createClass({
-  constructor: function() {
-    this.elementMap = {};
-    this.elementDetection = [];
-  },
-
-  /*
-   * Register a new element class for an element.
-   */
-  register: function(name, ElementClass) {
-    this.elementMap[name] = ElementClass;
-    return this;
-  },
-
-  /*
-   * Unregister a previously registered class for an element.
-   */
-  unregister: function(name) {
-    delete this.elementMap[name];
-    return this;
-  },
-
-  /*
-   * Add a new detection function to determine which element
-   * class to use when converting existing js instances into
-   * refract element.
-   */
-  detect: function(test, ElementClass, givenPrepend) {
-    var prepend = givenPrepend === undefined ? true : givenPrepend;
-
-    if (prepend) {
-      this.elementDetection.unshift([test, ElementClass]);
-    } else {
-      this.elementDetection.push([test, ElementClass]);
-    }
-
-    return this;
-  },
-
-  /*
-   * Convert an existing Javascript object into refract element instances, which
-   * can be further processed or serialized into refract or compact refract.
-   * If the item passed in is already refracted, then it is returned
-   * unmodified.
-   */
-  toElement: function(value) {
-    if (value instanceof BaseElement) { return value; }
-
-    var element;
-
-    for(var i = 0; i < this.elementDetection.length; i++) {
-      var test = this.elementDetection[i][0];
-      var ElementClass = this.elementDetection[i][1];
-
-      if (test(value)) {
-        element = new ElementClass(value);
-        break;
-      }
-    }
-
-    return element;
-  },
-
-  /*
-   * Get an element class given an element name.
-   */
-  getElementClass: function(element) {
-    var ElementClass = this.elementMap[element];
-
-    if (ElementClass === undefined) {
-      // Fall back to the base element. We may not know what
-      // to do with the `content`, but downstream software
-      // may know.
-      return BaseElement;
-    }
-
-    return ElementClass;
-  },
-
-  /*
-   * Convert a long-form refract document into refract element instances.
-   */
-   fromRefract: function(doc) {
-    var ElementClass = this.getElementClass(doc.element);
-    return new ElementClass().fromRefract(doc);
-  },
-
-  /*
-   * Convert a compact refract tuple into refract element instances.
-   */
-  fromCompactRefract: function(tuple) {
-    var ElementClass = this.getElementClass(tuple[0]);
-    return new ElementClass().fromCompactRefract(tuple);
+exports.init = function(registry) {
```

Comment:
I understand the use of `init` here but in many node.js modules they just export a default function. Also and perhaps more important, we should discuss how we plan to load the namespaces. Many, many node packages have a `.use(require('minim-api-description'))` or something like that. It's chainable and lets you load a bunch of plugins.